pkg/cache/clusterqueue_test.go

```diff
@@ -306,7 +381,32 @@ func TestFitInCohort(t *testing.T) {
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
 			defer features.SetFeatureGateDuringTest(t, features.LendingLimit, tc.enableLendingLimit)()
-			got := tc.cq.FitInCohort(tc.request)
+			cache := New(utiltesting.NewFakeClient())
+
+			for _, flavorName := range tc.flavorNames {
+				cache.AddOrUpdateResourceFlavor(utiltesting.MakeResourceFlavor(flavorName).Obj())
+			}
+
+			for _, cq := range tc.clusterQueue {
+				_ = cache.AddClusterQueue(context.Background(), cq)
+			}
+
+			snapshot := cache.Snapshot()
+
+			i := 0
+			for fr, v := range tc.usage {
+				admission := utiltesting.MakeAdmission("CQ")
+				quantity := resources.ResourceQuantity(fr.Resource, v)
+				admission.Assignment(fr.Resource, fr.Flavor, quantity.String())
+
+				wl := utiltesting.MakeWorkload(fmt.Sprintf("workload-%d", i), "default-namespace").ReserveQuota(admission.Obj()).Obj()
+
+				cache.AddOrUpdateWorkload(wl)
```

Comment:
Since we snapshotted above and don't snapshot again, this line is a no-op. Please remove

Reply:
fix: https://github.com/kubernetes-sigs/kueue/pull/2655/commits/6f4af1cd79e24761afcdc933ff203eaa9ee64856